src/TwitterOAuth.php

```diff
@@ -305,6 +305,49 @@ public function upload(
         }
     }
 
+    /**
+     * Upload media to api.x.com using X API v2.
+     *
+     * @param string $path
+     * @param array  $parameters
+     *
+     * @return array|object
+     */
+    public function uploadV2($path, array $parameters = [])
+    {
+        $init = $this->http('POST', self::API_HOST, $path, $this->mediaInitParametersV2($parameters), false);
+        // Append
+        $segmentIndex = 0;
+        $media = fopen($parameters['media'], 'rb');
```

Comment:
@jez321 
ファイルの存在チェックは不要そうでしょうか？

```suggestion
        if (!isset($parameters['media']) || !file_exists($parameters['media'])) {
             throw new \InvalidArgumentException('Media file does not exist');
        }
        $media = fopen($parameters['media'], 'rb');
```

既存実装（以下）でもやってないので、ここでは特に考慮不要ですかね？

https://github.com/socialdog-inc/twitteroauth/blob/e994af48698efb929fa4cecea93502abfb821eb9/src/TwitterOAuth.php#L424-L435

Reply:
確認ありがとうございます！そうですね、確かにチェックあっても良さそうですがテストがない＋もう少しで投稿v2がなくなるというのもあり既存の実装のままでも良いかなと少し思いました！